lib/schema.js

```diff
@@ -77,23 +77,73 @@ const processSchema = (
   }
 };
 
+const getCategoryType = (category) => {
+  const type = category.constructor.name;
+  return type === 'Object' ? 'Local' : type;
+};
+
+const verifyLink = (
+  source,
+  sourceName,
+  destination,
+  destinationName,
+  propertyName
+) => {
+  const sourceType = getCategoryType(source);
+  const destinationType = getCategoryType(destination);
+
+  if (destinationType === 'Log') {
+    return `Illegal reference to Log from ${sourceName}.${propertyName}`;
+  }
+
+  if (destinationType === 'Local' && sourceType !== 'Local') {
+    return `Illegal reference to Local category ${destination} from ` +
+      `${propertyName} ${sourceName}.${propertyName}`;
+  }
+
+  return null;
+};
+
 const linkSchemas = (
   // Crosslink loaded schema
   schemas // object, schema collection
 ) => {
+  const errors = [];
   for (const schemaName in schemas) {
     const schema = schemas[schemaName];
     for (const fieldName in schema) {
       const field = schema[fieldName];
       if (field.domain) {
         const domain = domains.get(field.domain);
-        if (domain) field.definition = domain;
+        if (domain) {
+          field.definition = domain;
+        } else {
+          errors.push(
+            `Domain '${field.domain}' referenced from ` +
+            `${schemaName}.${fieldName} is not defined`
+          );
+        }
       } else if (field.category) {
         const category = categories.get(field.category);
-        if (category) field.definition = category;
+        if (category) {
+          const error = verifyLink(
+            schema, schemaName, category, field.category, fieldName
+          );
+          if (!error) {
+            field.definition = category;
+          } else {
+            errors.push(error);
+          }
```

Comment:
in such cases I'd prefer:
```js
if (error) errors.push(error);
else field.definition = category;
```

Reply:
I don't like that style at all, moreover this way `else` has the same indentation as `else` one level higher, which is a bit confusing.
```js
if (category) {
  // ...
  if (!error) field.definition = category;
  else errors.push(error);
} else {
  // ...
}

```